notebooks/ch16/classification_cost_justification.md

```diff
@@ -0,0 +1,75 @@
+
+
+```python
+# HIDDEN
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+%matplotlib inline
+import ipywidgets as widgets
+from ipywidgets import interact, interactive, fixed, interact_manual
+
+sns.set()
+sns.set_context('talk')
+np.set_printoptions(threshold=20, precision=2, suppress=True)
+pd.options.display.max_rows = 7
+pd.options.display.max_columns = 8
+pd.set_option('precision', 2)
+# This option stops scientific notation for pandas
+# pd.set_option('display.float_format', '{:.2f}'.format)
+```
+
+## Approximating the Empirical Probability Distribution
+
+In this section, we introduce **KL divergence** and demonstrate how minimizing average KL divergence in binary classification is equivalent to minimizing average cross-entropy cost.
+
+Since logistic regression outputs probabilities, a logistic model produces a certain type of probability distribution. Specifically, it uses its parameters $ \hat \theta $ to estimate the probability that the label $ y $ is $ 1 $ for an example input $ x $.
+
+For example, suppose that $ x $ is a scalar recording the forecasted chance of rain for the day and $ y = 1 $ means that Mr. Doe takes his umbrella with him to work. A logistic model predicts the probability that Mr. Doe takes his umbrella given a forecasted chance of rain: $ \hat{P_\theta}(y = 1 | x) $.
+
+Collecting data on Mr. Doe's umbrella usage provides us with a method of constructing an empirical probability distribution $ P(y = 1 | x) $. For example, if there were five days where the chance of rain $ x = 0.60 $ and Mr. Doe only took his umbrella to work once, $ P(y = 1 | x = 0.60) = 0.20 $. We can compute a similar probability distribution for each value of $ x $ that appears in our data. Naturally, after fitting a logistic model we would like the distribution predicted by the model to be as close as possible to the empirical distribution from the dataset. That is, for all values of $ x $ that appear in our data, we want:
+
+$$ \hat{P_\theta}(y = 1 | x) \approx P(y = 1 | x) $$
+
+One commonly used metric to determine the "closeness" of two probability distributions is the Kullback–Leibler divergence, or KL divergence, which has its roots in information theory.
+
+## Defining Average KL Divergence
+
+KL Divergence reveals how imprecise the probability distribution output by a logistic model is relative to the data. For example, revisiting the basketball data, suppose we want to determine the probability that LeBron makes a shot 15 ft from the basket. KL divergence quantifies the difference between the probability output by our logistic model and the actual probability based on the dataset. The KL divergence for binary classification between two distributions $P$ and $\hat{P_\theta}$ for a single data point $(x, y)$ is given by:
```

Comment:
The second sentence doesn't seem to serve much purpose in this paragraph since it doesn't mention probability distributions at all. Did you mean to explain it further?

Otherwise, we might want to reorder the sentences in this paragraph a bit:

> KL divergence quantifies the difference between the probability distribution computed by our logistic model and the actual distribution based on the dataset. Intuitively, it calculates how imprecisely the logistic model estimates the distribution of labels in data.
> 
> The KL divergence for binary classification between two distributions $P$ and $\hat{P_\theta}$ for a single data point $(x, y)$ is given by: